### PR TITLE
[YUNIKORN-2783] Doc changes to describe the new behaviour and caveats

### PR DESCRIPTION
**Pull request by manirajv06**
### What is this PR for?

Doc changes to describe the new behaviour about queue name, user name and group names and caveats in labels.

### What type of PR is it?
* [ ] - Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2783

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: docs/user_guide/labels_and_annotations_in_yunikorn.md [hunk 1]
```diff
@@ -25,19 +25,21 @@ under the License.
...
+| `applicationId`                   | Associates this pod with an application.                                                                                                                                                                                                                                                                                                                                                                                                                |
+| `queue`                           | Selects the YuniKorn queue this application should be scheduled in. Queue name should comply with [Kubernetes Syntax and character set](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) and also with [Partition and Queue Configuration](queue_config#queues). If Queue name doesn't comply, Annotations could be used as described below. This may be ignored if a placement policy is in effect. |
```
Comment (chenyulin0719) [inline docs/user_guide/labels_and_annotations_in_yunikorn.md:33]: Note: The order has been changed in [[YUNIKORN-2504] Support canonical labels and align metadata retrieving order in shim ](https://issues.apache.org/jira/browse/YUNIKORN-2504) (Annotation queue prior to legacy label queue) .  It't not related to this PR. I will create another Jira to change it once the 1.6.0 related Jira under [YUNIKORN-2501](https://issues.apache.org/jira/browse/YUNIKORN-2501) are done.

File: docs/user_guide/queue_config.md [hunk 2]
```diff
@@ -143,7 +143,13 @@ The name of a queue must be unique at the level that the queue is defined.
 Since the queue structure is fully hierarchical queues at different points in the hierarchy may have the same name.
 As an example: the queue structure `root.testqueue` and `root.parent.testqueue` is a valid structure.
 A queue cannot contain a dot "." character as that character is used to separate the queues in the hierarchy.
-If the name is not unique for the queue in the configuration or contains a dot a parsing error is generated and the configuration is rejected. 
+
+Valid Queue name:
+
+* must be 64 characters or less,
+* could contain alphanumeric character ([a-z0-9A-Z]), underscores (_), colons (:), hashes (#), slashes (/), ampersands (&), and dashes (-).
```
Comment (chenyulin0719) [inline docs/user_guide/queue_config.md:150]: @manirajv06 no "ampersands (&)", should it be @?

> var QueueNameRegExp = regexp.MustCompile(`^[a-zA-Z0-9_:#/@-]{1,64}$`)
([Link](https://github.com/apache/yunikorn-core/blob/6939b13d1d0e65aef18722ef546413f3e1775be2/pkg/common/configs/configvalidator.go#L72C1-L73C1))

File: docs/user_guide/queue_config.md [hunk 3]
```diff
@@ -265,7 +271,19 @@ The star "*" is the wildcard character and matches all users or groups.
 Duplicate entries in the lists are ignored and do not cause a parsing error.
 Specifying a star beside other list elements is not allowed.
 When a wildcard group is configured, a limit must be configured with at least one named group.
-Parsing will reject the configuration with limits that do not follow this rule.
+
+Valid User name:
+
+* must start with lower case or upper case alphabet,
+* followed by alphanumeric character ([a-z0-9A-Z]), underscores (_), colons (:), hashes (#), slashes (/), ampersands (&), dots (.), and dashes (-).
```
Comment (chenyulin0719) [inline docs/user_guide/queue_config.md:278]: Ditto, no "ampersands (&)", should it be @?

> var UserRegExp = regexp.MustCompile(`^[_a-zA-Z][a-zA-Z0-9:#/_.@-]*[$]?$`)

([Link](https://github.com/apache/yunikorn-core/blob/6939b13d1d0e65aef18722ef546413f3e1775be2/pkg/common/configs/configvalidator.go#L77))

**Review by chenyulin0719 — CHANGES_REQUESTED** [review]
Found some incorrect descriptions.

**Review by chenyulin0719 — APPROVED** [review]
+1 LGTM

File: docs/user_guide/queue_config.md [hunk 4]
```diff
@@ -265,7 +271,19 @@ The star "*" is the wildcard character and matches all users or groups.
 Duplicate entries in the lists are ignored and do not cause a parsing error.
 Specifying a star beside other list elements is not allowed.
 When a wildcard group is configured, a limit must be configured with at least one named group.
-Parsing will reject the configuration with limits that do not follow this rule.
+
+Valid User name:
+
+* must start with lower case or upper case alphabet,
```
Comment (ryankert01) [inline docs/user_guide/queue_config.md:277]: can also starts with underscore (`^[_a-zA-Z]`)

File: docs/user_guide/queue_config.md [hunk 5]
```diff
@@ -265,7 +271,19 @@ The star "*" is the wildcard character and matches all users or groups.
...
+* followed by alphanumeric character ([a-z0-9A-Z]), underscores (_), colons (:), hashes (#), slashes (/), at signs (@), dots (.), and dashes (-).
+* could end with dollar ($).
+
+Valid Group name:
+
+* must start with lower case or upper case alphabet
```
Comment (ryankert01) [inline docs/user_guide/queue_config.md:283]: ditto.

**Review by ryankert01 — COMMENTED** [review]
Overall great! some nits.

File: docs/user_guide/queue_config.md (hunk 4 above)
Comment (ryankert01) [inline docs/user_guide/queue_config.md:277]:
```suggestion
* must start with a letter (uppercase or lowercase) or an underscore (_).
```

File: docs/user_guide/queue_config.md (hunk 5 above)
Comment (ryankert01) [inline docs/user_guide/queue_config.md:283]:
```suggestion
* must start with a letter (uppercase or lowercase) or an underscore (_).
```

File: docs/user_guide/labels_and_annotations_in_yunikorn.md (hunk 1 above)
Comment (pbacsko) [inline docs/user_guide/labels_and_annotations_in_yunikorn.md:33]: nit grammar/spelling: "if the queue doesn't comply, annotations could be used as described below"

File: docs/user_guide/labels_and_annotations_in_yunikorn.md [hunk 6]
```diff
@@ -25,19 +25,21 @@ under the License.
...
+| `spark-app-selector`              | Alternative method of specifying `applicationId` used by Spark Operator if the label `applicationId` and annotation `yunikorn.apache.org/app-id` unset.                                                                                                                                                                                                                                                                                                 |
```
Comment (pbacsko) [inline docs/user_guide/labels_and_annotations_in_yunikorn.md:34]: nit grammar: "...if the label `applicationId` and annotation `yunikorn.apache.org/app-id` are unset".

File: docs/user_guide/labels_and_annotations_in_yunikorn.md (hunk 1 above)
Comment (pbacsko) [inline docs/user_guide/labels_and_annotations_in_yunikorn.md:33]: Shouldn't we use the `<br>` tag here as well, just like below?

File: docs/user_guide/placement_rules.md [hunk 7]
```diff
@@ -72,6 +72,7 @@ This means that we allow user name and or tag values to contain dots without the
 For queues in the configuration that as an example must map to username with a dot you must specify them as follows:
 A user rule with the user `user.name` will generate the queue name `root.user_dot_name` as output.
 If that "user queue" must be added to the configuration the `user_dot_name` name should be used.
+Queue name should comply with rules described in [Partition and Queue Configuration](queue_config#queues).
```
Comment (pbacsko) [inline docs/user_guide/placement_rules.md:75]: nit: "The queue name"

File: docs/user_guide/queue_config.md [hunk 8]
```diff
@@ -143,7 +143,13 @@ The name of a queue must be unique at the level that the queue is defined.
 Since the queue structure is fully hierarchical queues at different points in the hierarchy may have the same name.
 As an example: the queue structure `root.testqueue` and `root.parent.testqueue` is a valid structure.
 A queue cannot contain a dot "." character as that character is used to separate the queues in the hierarchy.
-If the name is not unique for the queue in the configuration or contains a dot a parsing error is generated and the configuration is rejected. 
+
+Valid Queue name:
+
+* must be 64 characters or less,
+* could contain alphanumeric character ([a-z0-9A-Z]), underscores (_), colons (:), hashes (#), slashes (/), at signs (@), and dashes (-).
```
Comment (pbacsko) [inline docs/user_guide/queue_config.md:150]: nit: "could" -> "can"

File: docs/user_guide/queue_config.md [hunk 9]
```diff
@@ -143,7 +143,13 @@ The name of a queue must be unique at the level that the queue is defined.
...
+
+If the name is not unique for the queue in the configuration or contains a dot or do not follow the above rules a parsing error is generated and the configuration is rejected.
```
Comment (pbacsko) [inline docs/user_guide/queue_config.md:152]: nit: "...or does not follow the above rules, a parsing error is generated..."

File: docs/user_guide/queue_config.md [hunk 10]
```diff
@@ -265,7 +271,19 @@ The star "*" is the wildcard character and matches all users or groups.
 Duplicate entries in the lists are ignored and do not cause a parsing error.
 Specifying a star beside other list elements is not allowed.
 When a wildcard group is configured, a limit must be configured with at least one named group.
-Parsing will reject the configuration with limits that do not follow this rule.
+
+Valid User name:
+
+* must start with lower case or upper case alphabet,
+* followed by alphanumeric character ([a-z0-9A-Z]), underscores (_), colons (:), hashes (#), slashes (/), at signs (@), dots (.), and dashes (-).
+* could end with dollar ($).
```
Comment (pbacsko) [inline docs/user_guide/queue_config.md:279]: nit: "could" -> "can"

**Review by pbacsko — CHANGES_REQUESTED** [review]
Some nits

**Review by ryankert01 — APPROVED** [review]
+1, LGTM

**Review by pbacsko — APPROVED** [review]
LGTM

**Pull request closed without merging**